src/string.cr

```diff
@@ -2397,12 +2397,26 @@ class String
   #
   # "abcdef".compare("ABCDEF", case_insensitive: true) # => 0
   # "abcdef".compare("ABCDEG", case_insensitive: true) # => -1
+  #
+  # "heIIo".compare("heııo", case_insensitive: true, Unicode::CaseOptions::Turkic) # => 0
   # ```
-  def compare(other : String, case_insensitive = false)
+  def compare(other : String, case_insensitive = false, options = Unicode::CaseOptions::None)
     comparison = self <=> other
-    return comparison if (!case_insensitive || comparison != 0) && self.bytesize != other.bytesize
+    return comparison unless case_insensitive
+
+    if ascii_only? && other.ascii_only? && (options.none? || options.ascii?)
```

Comment:
I don't think `&& (options.none? || options.ascii?)` is needed here. Both strings are ASCII so we can safely take a shortcut ignoring the *options* value.

Reply:
Just wanted to prove you that it's needed but surprisingly you are right, thanks!